### PR TITLE
Fixes, tweaks and optimizations of Attack and Move actions

### PR DESCRIPTION
**Pull request by FS-21**
- Fixed 2 rare cases that made the team units unable to execute the script action at some point.
- Now the wait timer when the action didn't find a valid object won't be shared  with the GuardAreaTimer timer.
- pTeam->Focus will store (again) the selected team target in attack actions.
- Some tweaks.

### CODE REVIEW

**Comment by ayylmaoRotE** [comment]
Hoverattack (helicopters) are actually following their scripts and not being interrupted by attackers or when attacking. Nice work

File: src/Ext/Script/Body.cpp [hunk 1]
```diff
@@ -548,41 +551,43 @@ void ScriptExt::Mission_Attack(TeamClass *pTeam, bool repeatAction = true, int c
...
 	}
 
-	// This team has no units! END
+	// This team has no units!
 	if (!pTeam)
 	{
+		if (pTeamData->CloseEnough > 0)
+		{
+			pTeamData->CloseEnough = -1;
+		}
```
Comment (Thrifinesma) [inline src/Ext/Script/Body.cpp:573]:
```suggestion
		if (pTeamData->CloseEnough > 0)
			pTeamData->CloseEnough = -1;
```

File: src/Ext/Script/Body.cpp [hunk 2]
```diff
@@ -1000,35 +1023,42 @@ void ScriptExt::Mission_Attack(TeamClass *pTeam, bool repeatAction = true, int c
...
+						continue;
+					}
+
+					// Other cases
+					if (pUnitType->WhatAmI() != AbstractType::AircraftType)
+					{
+						if (pUnit->Target != pFocus)
+						{
+							pUnit->SetTarget(pFocus);
+						}
```
Comment (Thrifinesma) [inline src/Ext/Script/Body.cpp:1036]:
```suggestion
						if (pUnit->Target != pFocus)
							pUnit->SetTarget(pFocus);
```

File: src/Ext/Script/Body.cpp [hunk 3]
```diff
@@ -2008,42 +2038,44 @@ void ScriptExt::Mission_Move(TeamClass *pTeam, int calcThreatMode = 0, bool pick
...
-	// This team has no units! END
+	// This team has no units!
 	if (!pTeam)
 	{
-		auto pTeamData = TeamExt::ExtMap.Find(pTeam);
-		if (pTeamData && pTeamData->CloseEnough > 0)
+		if (pTeamData->CloseEnough > 0)
 		{
 			pTeamData->CloseEnough = -1;
 		}
```
Comment (Thrifinesma) [inline src/Ext/Script/Body.cpp:2074]:
```suggestion
			pTeamData->CloseEnough = -1;
```

**Review by Thrifinesma — CHANGES_REQUESTED** [review]

**Review by Metadorius — APPROVED** [review]

**Pull request merged**